spec/unit/resource/plist_spec.rb

```diff
@@ -0,0 +1,154 @@
+      end
+    end
+  end
+
+  describe "#convert_to_data_type_from_string" do
+    context "When the type is boolean and given a 1 or 0" do
+      it "returns true if entry is 1" do
+        expect(resource.convert_to_data_type_from_string("boolean", "1")).to eq true
+      end
+
+      it "returns false if entry is 0" do
+        expect(resource.convert_to_data_type_from_string("boolean", "0")).to eq false
+      end
+    end
+
+    context "When the type is integer and the value is 1" do
+      it "returns the value as an integer" do
+        expect(resource.convert_to_data_type_from_string("integer", "1")).to eq 1
+      end
+    end
+
+    context "When the type is integer and the value is 0" do
+      it "returns the value as an integer" do
+        expect(resource.convert_to_data_type_from_string("integer", "0")).to eq 0
+      end
+    end
+
+    context "When the type is integer and the value is 950224" do
+      it "returns the correct value as an integer" do
+        expect(resource.convert_to_data_type_from_string("integer", "950224")).to eq 950224
+      end
+    end
+
+    context "When the type is string and the value is also a string" do
+      it "returns the correct value still as a string" do
+        expect(resource.convert_to_data_type_from_string("string", "corge")).to eq "corge"
+      end
+    end
+
+    context "When the type is float and the value is 3.14159265359" do
+      it "returns the correct value as a float" do
+        expect(resource.convert_to_data_type_from_string("float", "3.14159265359")).to eq 3.14159265359
+      end
+    end
+
+    context "When the type nor the value is given" do
+      it "returns an empty string" do
+        expect(resource.convert_to_data_type_from_string(nil, "")).to eq ""
+      end
+    end
+  end
+
+  describe "#type_to_commandline_string" do
+    context "When given a certain data type" do
+      it "returns the required boolean entry type as a string" do
+        expect(resource.type_to_commandline_string(true)).to eq "bool"
+      end
+
+      it "returns the required array entry type as a string" do
+        expect(resource.type_to_commandline_string(%w{foo bar})).to eq "array"
+      end
+
+      it "returns the required dictionary entry type as a string" do
+        expect(resource.type_to_commandline_string("baz" => "qux")).to eq "dict"
+      end
+
+      it "returns the required string entry type as a string" do
+        expect(resource.type_to_commandline_string("quux")).to eq "string"
+      end
+
+      it "returns the required integer entry type as a string" do
+        expect(resource.type_to_commandline_string(1)).to eq "integer"
+      end
+
+      it "returns the required float entry type as a string" do
+        expect(resource.type_to_commandline_string(1.0)).to eq "float"
+      end
+    end
+  end
+end
```

Comment:
There are a lot of superfluous `context`s in here that don't really add much IMO.

Reply:
Nuked most of them